### ISSUE TITLE
Android notification not working

### ISSUE THREAD

**Issue by KeyvanPouyan** [xamarin/Monkey.Robotics#33]
Hey guys,
The device I am working with has several characteristics. There is one that we care about and enabling notification on it won't do much. I see that the onDescriptorWrite returns successful but no value is getting updated. I played around the Charactratics and Gattcallback but it seems nothing can make it work. I see that there are other users who have the same issue. I used the sample that you guys posted and notifications do not work as well.

**Comment by KevinPou** [comment]
Hey guys,
I managed to make it work. Apparently some BLE API implementation on Google side prefers having all the Gatt calls on the UI thread. Each call needs to be queued, so no overlapping calls. Each call needs to be completed before running the next one.

**Comment by KevinPou** [comment]
I did not contribute to the project as I understand having calls on the GUI thread is not desirable. Hopefully Android API gets a major facelift soon.
